docs/telemetry.md

```diff
@@ -1,55 +1,273 @@
 # Telemetry on Percona Server for MongoDB
 
-Percona telemetry fills in the gaps in our understanding of how you use Percona Server for MongoDB to improve our products. Participation in this anonymous program is optional. You can opt-out if you prefer to not share this information.
+Percona wants to learn more about how people use their software. This telemetry feature helps them understand what features users use the most, if there are any problems, and how Percona can improve their software in the future. 
+
+You can choose not to share this information. Participation in this program is completely voluntary. If you're uncomfortable sharing this anonymous data, you can [disable telemetry](#disable-telemetry).
+
+Your privacy is protected. Percona doesn't collect any personal information about you. All collected data is anonymous, meaning it can't be traced back to any individual user. To learn more about how Percona handles your data, read the [Percona Privacy statement](https://www.percona.com/privacy-policy).
+
+Currently, telemetry is added only to the Percona packages for both basic and Pro builds and to Docker images. 
 
 ## What information is collected
 
-Currently, telemetry is added only to the Percona packages and Docker images. It collects only information about the installation environment. Future releases may add additional telemetry metrics.
+Percona collects information using the following components:
+
+* The Telemetry subsystem collects the necessary metrics directly from the database and stores them in a Metrics file.
+
+* The Metrics file stores the metrics and is a standalone file located on the database host's file system.
+
+* The Telemetry Agent is an independent process running on your database host's operating system and carries out the following tasks:
+
+    * Collects OS-level metrics
+
+    * Reads the Metrics file, adds the OS-level metrics
+
+    * Sends the full set of metrics to the Percona Platform
+
+The telemetry also uses the Percona Platform with the following components:
+
+* Telemetry Service - offers an API endpoint for sending telemetry. The service handles incoming requests. This service saves the data into Telemetry Storage.
+
+* Telemetry Storage - stores all telemetry data for the long term.
+
+### Telemetry subsystem
+
+Telemetry subsystem extends the functionality of the database. It can be implemented in the following ways, depending on the database:
```

Comment:
```suggestion
The Telemetry subsystem extends the functionality of the database. It can be implemented in the following ways, depending on the database:
```